### DISCUSSION TITLE
Question on whether source code is BSL or Apache 2.0

### DISCUSSION THREAD

**Discussion by matthewmueller** [cockroachdb/cockroach#127140] [Q&A] [answered]
Hey folks, first of all, thanks for creating CockroachDB! As I'm reading through the codebase, I'm very impressed with how the abstractions have been built.

I'm interested in re-using the frontend aspect of Cockroach to build a query engine that can be used to query data from various SaaS providers. I'm planning on open sourcing this work, but I'm unsure which license it falls under. 

Reviewing the [license conversion timeline](https://www.cockroachlabs.com/docs/stable/licensing-faqs#license-conversion-timeline), am I understanding correctly:

- That the 21.1 release of Cockroach (including the source code at that point in time) is now Apache 2.0?
- Or is it the copyright at the top of the file + 3 years becomes Apache 2.0.

Specifically, I'd like to reuse [optgen](https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/opt/optgen/cmd/optgen/main.go), but I'm not sure if today's `optgen` is Apache 2.0 (because of the 2018 copyright) or if it's 21.1's version of optgen, or if it's still under BSL today.

Any clarification would be much appreciated!

(fwiw, I'm not opposed to getting the BSL license when the time comes, just way too early in development to consider something like that.)

**Comment by yuzefovich** [comment, marked as answer]
The first bullet is correct. The source code for the 21.1 release is now Apache 2.0. The dates in the copyright headers are not updated at every release.

**Comment by dt** [comment]
> the 21.1 release of Cockroach (including the source code at that point in time) is now Apache 2.0

If you're looking at optgen this shouldn't affect you, but note that this is only true for the portion of the codebase under the BSL; the sub-tree under pkg/ccl is under a different license (CCL) that does not transition to APL2 after a set duration.
